src/servers/main.cc

```diff
@@ -422,6 +429,28 @@ ParseFloatOption(const std::string arg)
   return std::stof(arg);
 }
 
+std::pair<int, std::vector<float>>
+ParseVGPUOption(const std::string arg)
```

Comment:
Do you check if the user does not follow the mentioned format: **"\<physical GPU\>;\<number of virtual GPUs\>;\<memory limit for physical GPU\>"**
You should check and report an appropriate error in this case. For example arg.find(";) may be -1 if there is no ";"

Reply:
To add to this, how does this handle the case where a user defines virtual GPUs on multiple physical GPUs?

Reply:
Hemant asked me about this as well. You just use the option multiple times. I'm currently adding this to the description of the option. Also changing the option name to "--tf-add-vgpu".